lib/topological_inventory/ansible_tower/operations/worker.rb

```diff
@@ -14,6 +16,11 @@ def initialize(messaging_client_opts = {})
         end
 
         def run
+          # Start up the receptor worker
+          receptor_client = ReceptorController::Client.new(:logger => logger)
+          TopologicalInventory::AnsibleTower::ConnectionManager.class_variable_set(:@receptor_client, receptor_client)
+          receptor_client.start
```

Comment:
I think we don't need new receptor_client for each request, it was intended as one client per pod (it means one kafka listener per process). The `logger` will be the same in both cases.
Then you don't need to propagate it to all the methods.

Reply:
:+1: pulled the receptor out of everything, the state is all handled in the ConnectionManager class.